metrics/metrics_client.go

```diff
@@ -0,0 +1,44 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	runtimemetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// Metrics subsystem and all of the keys used by the metrics client.
+const (
+	MetricsClientSubsystem    = "metrics_client"
+	MetricsClientFailTotalKey = "fail_total"
+)
+
+func init() {
+	registerMetricsClientMetrics()
+}
+
+type metricsClientFailTotalAdapter struct {
+	metric  *prometheus.CounterVec
+	address string
+}
+
+func (a *metricsClientFailTotalAdapter) Increment(query string) {
+	a.metric.WithLabelValues(a.address, query).Inc()
+}
+
+func (a *metricsClientFailTotalAdapter) SetAddress(address string) {
+	a.address = address
+}
+
+var (
+	metricsClientFailTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: MetricsNamespace,
+		Subsystem: MetricsClientSubsystem,
+		Name:      MetricsClientFailTotalKey,
+		Help:      "counter that indicates how many API requests to metrics server(e.g. prometheus) are failed.",
+	}, []string{"namespace", "name"})
```

Comment:
`namespace` and `name` labels are highly likely to conflict with other relabeled labels since it is too general. Would you add a prefix like `metrics_`?

Reply:
Correct my mistake, the label does not match the content.